MyAssignment2.Rmd

```diff
@@ -0,0 +1,174 @@
+---
+title: "MyAssignment2"
+author: "Elliot Hu-Au"
+date: "October 25, 2018"
+output: html_document
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE)
+```
+
+R Markdown for HUDK 4050: Assignment 2
+
+Part II
+```{r}
+
+
+E1<-dplyr::select(EDGE,count)
+student_count<- EDGE %>% group_by(EDGE$to) %>% summarize(sum = sum(count))
+names(student_count)[1]<-"id"
+new_count<- merge(VERTEX,student_count,by="id")
+
+g <- graph.data.frame(EDGE, directed=TRUE, vertices=all_totals)
+
+teach_count<- EDGE %>% group_by(from) %>% summarize(sum(count))
+names(teach_count)[2]<-"sum"
+names(teach_count)[1]<-"id"
+new_teach_count<- merge (VERTEX,teach_count, by="id")
+
+all_totals<-dplyr::bind_rows(new_teach_count,new_count)
+nodeSize<- setNames(c(all_totals$sum),c(all_totals$id))
+
+
+plot(g,layout=layout.fruchterman.reingold, vertex.color=all_totals$gender, edge.width=EDGE$count, vertex.size=all_totals$sum*3)
+```
+
+## Part III
+
+Now practice with data from our class. Please create a **person-network** with the data set hudk4050-classes.csv. To create this network you will need to create a person-class matrix using the tidyr functions and then create a person-person matrix using `t()`. You will then need to plot a matrix rather than a data frame using igraph.
+
+Once you have done this, also [look up](http://igraph.org/r/) how to generate the following network metrics: betweeness centrality and dregree. **Who is the most central person in the network?**
+
+
+```{r}
+
+class_2018<-read.csv("HUDK4050-2018-classes.csv")
+class_2018<-select(class_2018,"Q8", "Q9", "Q1", "Q3", "Q4", "Q5", "Q6")
+class_2018<- class_2018[-c(1,2,3,39),]
+class_2018<-unite(class_2018, "name", Q8,Q9, sep = " ", remove= TRUE)
+
+group_by(class_2018, Q1)
+class_2018%>% summarise(name)
+
+#creating the tall dataframe of names and classes
+names_class1<-select(class_2018, name, Q1)
+
+names_class2<-select(class_2018, name, Q3)
+names_class2<-names_class2[-c(1,39),]
+
+names_class3<-select(class_2018,name, Q4)
+names_class3<-arrange(names_class3, Q4)
+names_class3<-names_class3[-c(1:7),]
+
+names_class4<-select(class_2018,name, Q5)
+names_class4<-arrange(names_class4, Q5)
+names_class4<-names_class4[-c(1:12),]
+
+names_class5<-select(class_2018, name, Q6)
+names_class5<-arrange(names_class5, Q6)
+names_class5<-names_class5[-c(1:30),]
+
+names(names_class1)<-c("name","class")
+names(names_class2)<-c("name","class")
+names(names_class3)<-c("name","class")
+names(names_class4)<-c("name","class")
+names(names_class5)<-c("name","class")
+
```

Comment:
I think you may have taken the scenic route on this one. You could achieve the same outcome using a little less code: 

`D2 <- tidyr::gather(D1, course.label, course, Class.1, Class.2, Class.3, Class.4, Class.5, Class.6, na.rm = TRUE, convert = FALSE)`